### ISSUE TITLE
Circulation binary sensor missing (binary_sensor.dhw_circulation)

### ISSUE THREAD

**Issue by kalatos86** [signalkraft/mypyllant-component#132]
Hello,

just migrated my Vaillant VR920 from multimatic to myVaillant app. From the first look the future which I was using was the binary sensor: binary_sensor.dhw_circulation. Is there any chance the same one will be available through the myvaillant integration? I need this to control the external electric heater. Thanks!.

**Comment by PV-Joe** [comment]
I used the HACS mutimatic integration before and changed to HACS myVaillant as well and noticed that the sensor for circulation is missing.

My System is:
ecoTec plus with VR921

I think Vaillant is playing around with the API as other sensors are unavailable at the moment.

**Comment by kalatos86** [comment]
Definitely. Same for disabling the system. There is something in between like this away mode but this one not disabling hot water. Would be great if there will be possibility to re-create the service to disable the system. From the quick actions menu option is still available.

**Comment by signalkraft** [comment]
Could you describe what that sensor was doing? I don't have a circulation schedule in my setup. Was it just on when the circulation was active?

**Comment by kalatos86** [comment]
Yes. Wysłane z iPhone'aWiadomość napisana przez signalkraft ***@***.***> w dniu 18.03.2024, o godz. 14:57:﻿
Could you describe what that sensor was doing? I don't have a circulation schedule in my setup. Was it just on when the circulation was active?

**Comment by PV-Joe** [comment]
> Could you describe what that sensor was doing? I don't have a circulation schedule in my setup. Was it just on when the circulation was active?

… I don’t have circulation in my Hardware setup but had the circulation sensor. It showed circulation active in the morning.

Maybe you’ll get a hint for your myVaillant  integration from the multimatic integration:
https://github.com/thomasgermain/vaillant-component/blob/master/custom_components/multimatic/binary_sensor.py

**Comment by thomasgermain** [comment]
> Could you describe what that sensor was doing? I don't have a circulation schedule in my setup. Was it just on when the circulation was active?

Yeah indeed, the binary sensor was on when circulation was supposed to be active. I know circulation can have its own timeprogram. 
I never went any further because no one ever asked for something else than this binary sensor.

**Comment by signalkraft** [comment]
https://github.com/signalkraft/mypyllant-component/releases/tag/v0.7.4b10 should have a calendar entity for the circulation pump schedule. You should be able to use that to trigger your external heater, or other automations. It's not a drop-in replacement for the switch entity, but it's more flexible when you want to react slightly before or after the schedule for example.

**Comment by kalatos86** [comment]
Hello,

you mean the newest one? I am using mostly NodeRed for automation's and trying to get before the details from the entity: water_heater.kalatosy_domestic_hot_water_0 but the schedule is not readable in the logical way for me:


![image](https://github.com/signalkraft/mypyllant-component/assets/75856476/dbd562ec-28f6-41e0-bb29-5d776e067ab6)

Could you please advise how to dig into the details of circulation pump program schedule with the newest update?

Thanks!
Lukasz

> https://github.com/signalkraft/mypyllant-component/releases/tag/v0.7.4b10 should have a calendar entity for the circulation pump schedule. You should be able to use that to trigger your external heater, or other automations. It's not a drop-in replacement for the switch entity, but it's more flexible when you want to react slightly before or after the schedule for example.

**Comment by signalkraft** [comment]
I can't help with node red, in HA you would use calendar triggers: https://www.home-assistant.io/docs/automation/trigger/#calendar-trigger

I'm sure someone has written a template for turning this into a binary switch, too, in the HA forums!

**Comment by kalatos86** [comment]
it's probably too much for me, I can't do it this way :) The question is whether the creators of this customization might consider adding such a binary switch in future versions

**Comment by boelle** [comment]
@signalkraft did you add such a switch ?

**Comment by boelle** [comment]
@signalkraft poke
